### PR TITLE
Filter out blocked regions from possible recruitment regions

### LINKED ISSUE #418: Users can muster ships into blocked regions
Screenshot_2020-03-25_at_12 07 22

### PR DESCRIPTION
**Pull request by gereon77**
Closes #418

### CODE REVIEW

File: agot-bg-game-server/src/common/ingame-game-state/westeros-game-state/mustering-game-state/player-mustering-game-state/PlayerMusteringGameState.ts [hunk 1]
```diff
@@ -281,7 +281,9 @@ export default class PlayerMusteringGameState extends GameState<ParentGameState>
                 .filter(r => r.type.kind == RegionKind.SEA)
             )
             // Can't recruit in a enemy-controled territory
-            .filter(r => r.getController() == null || r.getController() == this.house);
+            .filter(r => r.getController() == null || r.getController() == this.house)
+            // Can't recruit into a blocked region, e.g. an adjacent blocked sea
+            .filter(r => r.garrison != 1000);
```
Comment (Longwelwind) [inline agot-bg-game-server/src/common/ingame-game-state/westeros-game-state/mustering-game-state/player-mustering-game-state/PlayerMusteringGameState.ts]: Why not simply `r.garrison > 0`?

Comment (gereon77) [reply agot-bg-game-server/src/common/ingame-game-state/westeros-game-state/mustering-game-state/player-mustering-game-state/PlayerMusteringGameState.ts]: Then you wouldnt be able to muster in your capital anymore :)

Comment (Longwelwind) [reply agot-bg-game-server/src/common/ingame-game-state/westeros-game-state/mustering-game-state/player-mustering-game-state/PlayerMusteringGameState.ts]: Ah, true that. Maybe `r.garrison > 0 ? r.getController() == this.house : true`.

It's a small change, but it's to support possible neutral garrisons on seas. On top of that, I'd prefer to make the `1000` magic value as rare as possible in the codebase. 🙂

Comment (gereon77) [reply agot-bg-game-server/src/common/ingame-game-state/westeros-game-state/mustering-game-state/player-mustering-game-state/PlayerMusteringGameState.ts]: Ok np. I will change that after I rebased the other PRs

**Pull request merged**